### PR TITLE
ci: gen_dox: update uncommon driver path handling

### PR DESCRIPTION
**Pull request by amiclaus**
## Pull Request Description

Create a list and pattern of uncommon driver paths than need special handling when generating the doxygen drivers ToC page.

Remove multiple if statements and replace them with a list that cand be easily extended when needed.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [ ] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

**Review by rbolboac — APPROVED** [review]

File: ci/gen_dox.sh [hunk 1]
```diff
@@ -35,6 +35,12 @@
 #List of excluded driver folders for the documentation generation.
 EXCLUDE_DRV="talise navassa ad9083_api ad5940 madura"
 
+#List of folders that are handled differently due to their uncommon structure
+uncommon_drv_list=("sd-card" "imu" "api" "tdm")
```
Comment (buha) [inline ci/gen_dox.sh:39]: tdm should go under api, i can do that

Comment (buha) [reply ci/gen_dox.sh:39]: I made #2073, you can remote tdm from the list

Comment (amiclaus) [reply ci/gen_dox.sh:39]: Yep, the tdm folder didn't make sense. Thanks. Will update the PR also.

**Review by amiclaus — COMMENTED** [review]
v2:
- drop `tdm` folder from the list, since the content was moved under `api` folder.

**Review by buha — APPROVED** [review]

**Pull request merged**